parallel-install/pkg/helm/client.go

```diff
@@ -17,22 +21,46 @@ import (
 
 const logPrefix = "[helm/client.go]"
 
+//Config provides configuration for the Client
 type Config struct {
-	HelmTimeoutSeconds            int
-	BackoffInitialIntervalSeconds int
-	BackoffMaxElapsedTimeSeconds  int
-	Log                           func(format string, v ...interface{})
+	HelmTimeoutSeconds            int                                   //Underlying native Helm client processing timeout
+	BackoffInitialIntervalSeconds int                                   //Initial interval for exponential-backoff retry algorithm
+	BackoffMaxElapsedTimeSeconds  int                                   //Maximum time for exponential-backoff retry algorithm
+	Log                           func(format string, v ...interface{}) //Used for logging
 }
 
+//Client type implements ClientInterface
 type Client struct {
 	cfg Config
 }
 
+//ClientInterface defines the contract for the helm-related installation processes.
 type ClientInterface interface {
+	//InstallRelease installs a named chart from local filesystem directory with specific overrides.
+	//The function retries on errors according to Config provided to the Client.
+	//
+	//ctx is used for cancellation of the operation.
+	//Cancellation of the successfull operation is not possible,
+	//because the underlying Helm operation are blocking and do not support Context-based cancellation.
+	//Cancellation is possible when errors occur and the operation is re-tried.
+	//When operation is re-tried, it is not guaranteed that cancellation is handled immediately, due to the blocking nature of Helm client calls.
+	//However, once the underlying Helm operations ends, the cancel condition is detected and the operation's result is returned without further retries.
 	InstallRelease(ctx context.Context, chartDir, namespace, name string, overrides map[string]interface{}) error
+	//UninstallRelease uninstalls a named chart from the cluster.
+	//The function retries on errors according to Config provided to the Client.
+	//
+	//ctx is used for cancellation of the operation.
+	//Cancellation of the successfull operation is not possible,
+	//because the underlying Helm operation are blocking and do not support Context-based cancellation.
+	//Cancellation is possible when errors occur and the operation is re-tried.
+	//When operation is re-tried, it is not guaranteed that cancellation is handled immediately, due to the blocking nature of Helm client calls.
+	//However, once the underlying Helm operations ends, the cancel condition is detected and the operation's result is returned without further retries.
 	UninstallRelease(ctx context.Context, namespace, name string) error
 }
 
+//NewClient returns a new Client instance.
+//If you need different configuration for installation and uninstallation,
```

Comment:
User is not able to use his own instance of Client to start installation/uninstallation. He can use different Installation, though.

Reply:
Because the type is public I documented it from the type's perspective, not how it's used in top-level API. Leaving as it is.